extension-style/src/main/java/com/mapbox/maps/extension/style/types/PromoteId.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.maps.extension.style.types
+
+import androidx.annotation.Keep
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.extension.style.sources.generated.GeoJsonSource
+import com.mapbox.maps.extension.style.sources.generated.VectorSource
+
+/**
+ * Holds a property type to promote a specific feature for feature state API.
+ *
+ * @param sourceId source layer id of the feature, either source [GeoJsonSource] or [VectorSource].
```

Comment:
Do we have a check for that?

Reply:
`promoteId` is only available for GeoJson or vector source on [builder method](https://github.com/mapbox/mapbox-maps-android/pull/636/files#diff-f5e000d4ded37a7719c6f0253cb88496cf0820543ec87717a2eed34dfd169eadR586) so we don't need to check here.

But if you are asking that if we initialize promoteId with the wrong sourceId and provide that to geojson builder -> IMO it should be handled by core itself, and shouldn't be applied to feature state, but I will test it.